@xen-orchestra/proxy/src/app/_debounceResource.js

```diff
@@ -12,24 +14,27 @@ export async function debounceResource(pDisposable, hooks, delay = 0) {
     if (timeoutId !== undefined) {
       clearTimeout(timeoutId)
       timeoutId = undefined
-      hooks.removeListener('stop', onStop)
 
       return disposable.dispose()
     }
   }
 
-  const onStop = () => {
+  disposers.add(() => {
     const shouldDisposeNow = timeoutId !== undefined
     if (shouldDisposeNow) {
       return disposeWrapper()
     } else {
       // will dispose ASAP
       delay = 0
     }
-  }
-  hooks.on('stop', onStop)
+  })
 
   return new Disposable(disposable.value, () => {
     timeoutId = setTimeout(disposeWrapper, delay)
   })
 }
+debounceResource.flushAll = () =>
+  asyncMap(disposers, dispose => {
+    disposers.delete(dispose)
```

Comment:
You can use `Set#clear` instead of deleting each entry one by one.

Reply:
What if a disposer is added on processing the disposers?

Reply:
You current code don't handle it either.

Also, what if one is added after? It's not the proper way to handle this.

Reply:
It will be properly cleaned in the next flush.